docs.go

```diff
@@ -0,0 +1,56 @@
+// Package artifact provides an interface for working with the Taskcluster
+// Queue's blob artifacts.  Blob artifacts are the way that Taskcluster stores
+// and distributes the results of a task and replace the old "S3" type of
+// artifact for storing artifacts in S3.  These artifacts have stronger
+// authenticity and integrity guaruntees than the old type.
+//
+// Overview of Blob Artifacts
+//
+// Blob artifacts can be between 1 byte and 5GB if uploaded as a single part
+// upload and between 1 byte and 5TB if uploaded as a multipart upload.  To
+// upload this type of artifact, the uploader must compute the artifacts sha256
+// and size before and after optional gzip compression.  The sha256 and size
+// values are used by the Queue to generate a set of requests which get sent
+// back to the uploader which can be used to upload the artifact.  This ensures
+// that network interuptions or corruption result in errors when uploading.
+// Once the uploads have completed, the uploader must tell the Queue that the
+// upload is complete.
+//
+// The queue ensures that the sha256 and size values are set as headers on the
+// artifacts in S3 so that downloaders can do verification
```

Comment:
missing '.' at end of sentence - although I prefer the alternative ending " .... so that downloaded content can be verified."